### PR TITLE
Clear webhook cache when webhooks are deleted

### LINKED ISSUE #689: Deleted Webhook still triggers
I think its due to forever caching
https://github.com/pelican-dev/panel/blob/3f9c1dbc3c3438f17ab09aa5b8d8f2362eec8efc/app/Models/WebhookConfiguration.php#L44
https://github.com/pelican-dev/panel/blob/3f9c1dbc3c3438f17ab09aa5b8d8f2362eec8efc/app/Listeners/DispatchWebhooks.php#L16

### PR DESCRIPTION
**Pull request by lancepioch**
Fixes #689

### CODE REVIEW

File: app/Models/WebhookConfiguration.php [hunk 1]
```diff
@@ -40,12 +40,21 @@ protected static function booted(): void
...
             ])->unique();
 
-            $changedEvents->each(function (string $event) {
-                cache()->forever("webhooks.$event", WebhookConfiguration::query()->whereJsonContains('events', $event)->get());
-            });
+            self::updateCache($changedEvents);
+        });
+
+        self::deleted(static function (self $webhookConfiguration): void {
+            self::updateCache(collect($webhookConfiguration->events));
```
Comment (RMartinOscar) [inline app/Models/WebhookConfiguration.php:47]:
```suggestion
            self::updateCache((array) collect($webhookConfiguration->events));
```

File: app/Models/WebhookConfiguration.php [hunk 2]
```diff
@@ -40,12 +40,21 @@ protected static function booted(): void
...
+        });
+    }
 
-            cache()->forever('watchedWebhooks', WebhookConfiguration::pluck('events')->flatten()->unique()->values()->all());
+    private static function updateCache($eventList): void
```
Comment (RMartinOscar) [inline app/Models/WebhookConfiguration.php:51]:
```suggestion
    private static function updateCache(Collection $eventList): void
```

File: app/Models/WebhookConfiguration.php (hunk 1 above)
Comment (RMartinOscar) [inline app/Models/WebhookConfiguration.php:47]:
```suggestion
            self::updateCache((array) collect($webhookConfiguration->events));
```

Comment (RMartinOscar) [inline app/Models/WebhookConfiguration.php:47]:
```suggestion
            self::updateCache(collect((array) $webhookConfiguration->events));
```

File: app/Models/WebhookConfiguration.php (hunk 2 above)
Comment (RMartinOscar) [inline app/Models/WebhookConfiguration.php:51]:
```suggestion
    private static function updateCache(Collection $eventList): void
```

**Review by RMartinOscar — CHANGES_REQUESTED** [review]
Add this on top
```php
use Illuminate\Support\Collection;
```

**Review by RMartinOscar — APPROVED** [review]

**Review by notAreYouScared — APPROVED** [review]

**Pull request merged**